subgraph/subgraph.template.yaml

```diff
@@ -0,0 +1,68 @@
+specVersion: 0.0.4
+schema:
+  file: ./schema.graphql
+dataSources:
+  - kind: ethereum/contract
+    name: OrderBook
+    network: {{ network }}
+    source:
+      address: "{{ orderbook }}"
+      abi: OrderBook
+      startBlock: {{ blockNumber }}
```

Comment:
can we do yaml manipulation as structured data rather than with string interpolation?

Reply:
Ok, I'm using `serde_yaml` to manipulate the subgraph template from rust (i built a cli to do this aswell) and modify each field.

The issue that I had was handling the address present in the YAML file since it was not possible for me to put the address with quotes using just serde_yaml. This because serde_yaml put the address as a hex value without quotes.

I tried using the `format` macro to put the quotes but then extra quotes are added by the serialization - if I use single quotes, then the output give me 3 single quotes, and if I use double quotes, then I get single quote + double quote surrounded the address. 

So, I ended using `replace` on the string before writing the file

Reply:
I'll mark as resolved since this code is now part of other repo: [subgraph-cli](https://github.com/rainprotocol/rain.subgraph-cli) which is used inside the nix flake,